x/metadata/keeper/msg_server.go

```
@@ -411,3 +413,112 @@ func (k msgServer) DeleteRecordSpecification(
+	// Validate
+	if err := msg.ValidateBasic(); err != nil {
+		ctx.Logger().Error("unable to validate message", "err", err)
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, err.Error())
+	}
+
+	if k.Keeper.OSLocatorExists(ctx, msg.Locator.Owner) {
+		ctx.Logger().Error("Address already bound to an URI", "owner", msg.Locator.Owner)
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, types.ErrOSLocatorAlreadyBound.Error())
+	}
+
+	// Bind owner to URI
+	address, err := sdk.AccAddressFromBech32(msg.Locator.Owner)
+	if err != nil {
+		ctx.Logger().Error("invalid address", "err", err)
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, err.Error())
+	}
+
+	if err := k.Keeper.SetOSLocatorRecord(ctx, address, msg.Locator.LocatorUri); err != nil {
+		ctx.Logger().Error("unable to bind name", "err", err)
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, err.Error())
+	}
+	// Emit event and return
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(
+			types.EventTypeOsLocatorCreated,
+			sdk.NewAttribute(types.AttributeKeyOSLocatorAddress, msg.Locator.Owner),
+			sdk.NewAttribute(types.AttributeKeyOSLocatorURI, msg.Locator.LocatorUri),
+		),
+	)
+
+	return &types.MsgBindOSLocatorResponse{}, nil
+}
+
+func (k msgServer) DeleteOSLocator(ctx context.Context, msg *types.MsgDeleteOSLocatorRequest) (*types.MsgDeleteOSLocatorResponse, error) {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	// Validate
+	if err := msg.ValidateBasic(); err != nil {
+		sdkCtx.Logger().Error("unable to validate message", "err", err)
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, err.Error())
+	}
+
+	if !k.Keeper.OSLocatorExists(sdkCtx, msg.Locator.Owner) {
+		sdkCtx.Logger().Error("Address not already bound to an URI", "owner", msg.Locator.Owner)
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, types.ErrOSLocatorAlreadyBound.Error())
+	}
+
+	// already valid address, checked in ValidateBasic
+	ownerAddr, _ := sdk.AccAddressFromBech32(msg.Locator.Owner)
```

Ignored error?  At least add a comment why this error isn't returned.